presto-main/src/test/java/com/facebook/presto/cost/TestFilterStatsRule.java

```diff
@@ -0,0 +1,292 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.cost;
+
+import com.facebook.presto.sql.planner.Symbol;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.sql.planner.iterative.rule.test.PlanBuilder.expression;
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+
+public class TestFilterStatsRule
+        extends BaseStatsCalculatorTest
+{
+    public StatsCalculatorTester defaultFilterTester;
+
+    @BeforeClass
+    public void setupClass()
+    {
+        defaultFilterTester = new StatsCalculatorTester(
+                testSessionBuilder()
+                        .setSystemProperty("default_filter_factor_enabled", "true")
+                        .build());
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDownClass()
+    {
+        defaultFilterTester.close();
+        defaultFilterTester = null;
+    }
+
+    @Test
+    public void testEstimatableFilter()
```

Comment:
That's a lot of tests here. 
You're adding `TestFilterStatsRule` while the existing tests are in `TestFilterStatsCalculator`.
The difference is subtle -- Rule applies default factor. For the rest, it delegates to `FilterStatsCalculator`
Hence, it should be sufficient to have a simple test that covers default factor application only.

Reply:
I'll get rid of one of the tests.  but i think having 1 positive and 1 negative test is useful.